src/main.c

```diff
@@ -23,8 +23,8 @@ extern void __orig___schedule(bool);
 int process_id[MAX_CPU_NR];
 atomic_t cpu_finished;
 atomic_t clear_finished;
+atomic_t redirect_finished;
```

Comment:
1. Both patch 1 and patch 2 are used to fix same issue, it's better to merge them into one. 
2. commit log should be improved.   w/o sync ---> then using functions from old scheduler.